src/anomalib/utils/callbacks/thresholding.py

```diff
@@ -0,0 +1,159 @@
+        if "mask" in outputs.keys() and "anomaly_maps" in outputs.keys():
+            pl_module.pixel_threshold.cpu()
+            pl_module.pixel_threshold.update(outputs["anomaly_maps"], outputs["mask"].int())
+
+    def _reset(self, pl_module: AnomalyModule) -> None:
+        pl_module.image_threshold.reset()
+        pl_module.pixel_threshold.reset()
+
+    def _compute(self, pl_module: AnomalyModule) -> None:
+        pl_module.image_threshold.compute()
+        if pl_module.pixel_threshold._update_called:
+            pl_module.pixel_threshold.compute()
+        else:
+            pl_module.pixel_threshold.value = pl_module.image_threshold.value
+
+    def _outputs_to_cpu(self, output: STEP_OUTPUT):
+        if isinstance(output, dict):
+            for key, value in output.items():
+                output[key] = self._outputs_to_cpu(value)
+        elif isinstance(output, Tensor):
+            output = output.cpu()
+        return output
+
+    def _initialize_thresholds(
+        self,
+        threshold: BaseAnomalyThreshold
+        | tuple[BaseAnomalyThreshold, BaseAnomalyThreshold]
+        | DictConfig
+        | ListConfig
+        | str,
+    ) -> None:
+        if isinstance(threshold, BaseAnomalyThreshold):
+            self.image_threshold = threshold
+            self.pixel_threshold = threshold
+        elif isinstance(threshold, tuple):
+            self.image_threshold = threshold[0]
+            self.pixel_threshold = threshold[1]
+        elif isinstance(threshold, (str, DictConfig, ListConfig)):
+            self._load_from_config(threshold)
+        else:
+            raise ValueError(f"Invalid threshold type {type(threshold)}")
+
+    def _load_from_config(self, threshold: DictConfig | str | ListConfig) -> None:
+        """Loads the thresholding class based on the config.
+
+        Example:
+            threshold: F1AdaptiveThreshold
+            or
+            threshold:
+                class_path: F1AdaptiveThreshold
+                init_args:
+                    -
+            or
+            threshold:
+                - adaptive
+                - adaptive
+            or
+            threshold:
+                - class_path: F1AdaptiveThreshold
+                    init_args:
+                        -
+                - class_path: F1AdaptiveThreshold
+        """
+        if isinstance(threshold, (str, DictConfig)):
+            self.image_threshold = self._get_threshold_from_config(threshold)
+            self.pixel_threshold = self.image_threshold.clone()
+        elif isinstance(threshold, ListConfig):
+            self.image_threshold = self._get_threshold_from_config(threshold[0])
+            self.pixel_threshold = self._get_threshold_from_config(threshold[1])
+        else:
+            raise ValueError(f"Invalid threshold config {threshold}")
+
+    def _get_threshold_from_config(self, threshold: DictConfig | str) -> BaseAnomalyThreshold:
+        if isinstance(threshold, str):
```

Comment:
This function seems quite important. 

Can you add an `Example section here for a future reference? Like,

```python
def _get_threshold_from_config(self, threshold: DictConfig | str) -> BaseAnomalyThreshold:
    """
    Example:
         >>> _get_threshold_from_config(threshold="adaptive")
         ...
    """
```